### PR TITLE
chore: merge cosmos-sdk 0.45.11

### PR DESCRIPTION
**Pull request by JimLarson**
## Description

Refs: Agoric/agoric-sdk#6305

Merges cosmos-sdk 0.45.11, upgrade tendermint dependency to 0.34.23 (Agoric fork).

### Author Checklist

I have...

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x] added `!` to the type prefix if API or client breaking change
- [x] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] provided a link to the relevant issue or specification
- [x] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules)
- [x] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#testing)
- [x] added a changelog entry to `CHANGELOG.md`
- [x] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [x] updated the relevant documentation or specification
- [x] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by JimLarson** [comment]
The following files required manual action to resolve the merge, build, and pass tests:

```
        both modified:   baseapp/abci.go
        both modified:   baseapp/abci_test.go
        both modified:   contrib/images/simd-env/Dockerfile
        both modified:   go.mod
        both modified:   go.sum
        both added:      ics23/go/go.mod
        both added:      ics23/go/go.sum
        both modified:   x/auth/keeper/keeper.go
        both modified:   x/auth/vesting/types/common_test.go
        both modified:   x/distribution/keeper/delegation.go
```

File: Makefile [hunk 1]
```diff
@@ -400,7 +400,7 @@ proto-check-breaking:
 	@$(DOCKER_BUF) breaking --against $(PR_TARGET_REPO)#branch=$(PR_TARGET_BRANCH)
 
 
-TM_URL              = https://raw.githubusercontent.com/tendermint/tendermint/v0.34.0-rc6/proto/tendermint
+TM_URL              = https://raw.githubusercontent.com/tendermint/tendermint/v0.34.22/proto/tendermint
```
Comment (michaelfig) [inline Makefile]: Should point to our fork?

```suggestion
TM_URL              = https://raw.githubusercontent.com/agoric-labs/tendermint/v0.34.23-alpha.agoric.2/proto/tendermint
```

**Review by michaelfig — APPROVED** [review]
LGTM!  One little nit wrt our Tendermint proto source URL.  Feel free to address or not based on deadlines.

**Comment by JimLarson** [comment]
I updated both this and the `confio` version. Note that the `proto-update-deps` target is broken on v0.45.11, so this change is unlikely to matter.

**Pull request merged**